Datadog.Serverless.Compat/DatadogServerlessCompat.cs

```diff
@@ -0,0 +1,184 @@
+                { "ERROR", LogLevel.Error },
+                { "WARN", LogLevel.Warning },
+                { "INFO", LogLevel.Information },
+                { "DEBUG", LogLevel.Debug },
+                { "TRACE", LogLevel.Trace }
+            };
+
+            LogLevel logLevel = LogLevel.Information;
+            if (
+                !string.IsNullOrEmpty(logLevelEnv)
+                && logLevelMapping.TryGetValue(logLevelEnv, out var mappedLevel)
+            )
+            {
+                logLevel = mappedLevel;
+            }
+
+            var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddConsole().SetMinimumLevel(logLevel);
+            });
+            _logger = loggerFactory.CreateLogger("Datadog.Serverless.Compat");
+        }
+
+        private static bool IsWindows()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        }
+
+        private static bool IsLinux()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+        }
+
+        private static CloudEnvironment GetEnvironment()
+        {
+            var env = Environment.GetEnvironmentVariables();
+
+            if (
+                env.Contains("FUNCTIONS_EXTENSION_VERSION")
+                && env.Contains("FUNCTIONS_WORKER_RUNTIME")
+            )
+            {
+                return CloudEnvironment.AzureFunction;
+            }
+
+            if (env.Contains("FUNCTION_NAME") && env.Contains("GCP_PROJECT"))
+            {
+                return CloudEnvironment.GoogleCloudRunFunction1stGen;
+            }
+
+            return CloudEnvironment.Unknown;
+        }
+
+        private static string GetBinaryPath()
+        {
+            if (!string.IsNullOrEmpty(BinaryPath))
+            {
+                _logger.LogDebug("Detected user configured binary path {BinaryPath}", BinaryPath);
+                return BinaryPath;
+            }
+
+            if (IsWindows())
+            {
+                _logger.LogDebug("Detected {OS}", OS);
+                return "datadog/bin/windows-amd64/datadog-serverless-compat.exe";
+            }
+            else
+            {
+                _logger.LogDebug("Detected {OS}", OS);
+                return "datadog/bin/linux-amd64/datadog-serverless-compat";
+            }
+        }
+
+        private static string SetPackageVersion()
+        {
+            try
+            {
+                var assembly = Assembly.GetExecutingAssembly();
+                var version = assembly
+                    .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                    ?.InformationalVersion;
+                return version ?? "unknown";
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Unable to identify package version");
+                return "unknown";
+            }
+        }
+
+        public static void Start()
```

Comment:
Are users expected to call `DatadogServerlessCompat.Start()` themselves?

Reply:
Yes that's the current design. Do you agree with this approach?

Reply:
It's just that we generally get pushback from users about making any code changes in their apps.